lightly/loss/wmse_loss.py

```diff
@@ -59,10 +61,18 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
 
         f_cov_shrinked = (1 - self.eps) * f_cov + self.eps * eye
 
+        # get type of f_cov_shrinked and temporary convert to full precision
+        # to support chelosky decomposition
+        f_cov_shrinked_type = f_cov_shrinked.dtype
```

Comment:
This looks super duper hacky. Why is it necessary?

Reply:
Yes, as written in the comment. The original code is not using half precision.